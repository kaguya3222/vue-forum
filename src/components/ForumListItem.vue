<template>
  <div class="forum-listing">
    <div class="forum-details">
      <router-link
        class="text-xlarge"
        :to="{ name: 'Forum', params: { id: forum['.key'] } }"
      >
        {{ forum.name }}
      </router-link>
      <p>{{ forum.description }}</p>
    </div>

    <div class="threads-count">
      <p class="count">{{ threadsCount }}</p>
      {{ threadsCountWord }}
    </div>

    <div class="last-thread">
      <!-- TODO: implement later -->
    </div>
  </div>
</template>

<script>
export default {
  props: {
    forum: {
      required: true,
      type: Object
    }
  },
  name: "ForumListItem",
  computed: {
    threadsCount() {
      return this.forum.threads ? Object.values(this.forum.threads).length : 0;
    },
    threadsCountWord() {
      return this.threadsCount === 1 ? "thread" : "threads";
    }
  }
};
</script>

<style scoped></style>
