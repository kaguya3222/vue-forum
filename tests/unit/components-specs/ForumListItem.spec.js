import { mount } from "@vue/test-utils";
import ForumListItem from "../../../src/components/ForumListItem";
import mockedSourceData from "../mocks/mockedSourceData";

describe("ForumListItem", () => {
  const wrapper = mount(ForumListItem, {
    propsData: {
      forum: mockedSourceData.forums[0]
    },
    stubs: {
      "router-link": true
    }
  });
  test("Shows forum data", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Counts threads number", () => {
    const threads = Object.values(mockedSourceData.forums[0].threads);
    expect(wrapper.vm.threadsCount).toBe(threads.length);
  });
});
