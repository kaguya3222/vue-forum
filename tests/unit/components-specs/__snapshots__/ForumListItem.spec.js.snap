// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumListItem Shows forum data 1`] = `
<div class="forum-listing">
  <div class="forum-details">
    <router-link-stub to="[object Object]" class="text-xlarge">
      Announcements
    </router-link-stub>
    <p>Important announcements about the forum.</p>
  </div>
  <div class="threads-count">
    <p class="count">1</p>
    thread
  </div>
  <div class="last-thread"></div>
</div>
`;
